### PR TITLE
New feature: Email when a rare pokemon is found

### PR DESCRIPTION
**Pull request by mikehow1984**
This pull request includes a

- [ ] Bug fix
- [x] New feature
- [ ] Translation

The following changes were made

-added method to models that lets you use your GMail to mail yourself when a rare Pokemon is found
-the credentials json now includes gmail name, gmail password, and rare pokemon fields (a list of pokemon you want notifications for)
-made it so that credentials now load correctly in the load_credentials method

### CODE REVIEW

File: pogom/utils.py [hunk 1]
```diff
@@ -101,7 +101,7 @@ def get_pokemon_name(pokemon_id):
     return get_pokemon_name.names[str(pokemon_id)]
 
 def load_credentials(filepath):
-    with open(filepath+os.path.sep+'credentials.json') as file:
+    with open(filepath+os.path.sep+'credentials.example.json') as file:
```
Comment (MonkeyLeeT) [inline pogom/utils.py]: Nice addition. Could you leave the credentials.json line as it was here? We use example specifically just as example. No need to really use it.

Comment (patrickread) [inline pogom/utils.py]: Don't use the example file in your code. Just copy it to the `credentials.json` location for use:
```
cp credentials.example.json credentials.json
```

**Comment by prajwalkman** [comment]
Should it need a password? Can't we just ust smtp to send a mail with an arbitrary (hardcoded) "from" address without having to create a gmail account?

**Comment by patrickread** [comment]
@prajwalkman If it does need a password, you can always use an app specific password on this. Or, if this was a deployed application, it'd be an OAuth connection anyway.

**Comment by pickett2k** [comment]
could anyone explain how i implement these? i have overwritten the existing files with the files changed but it still doesnt work? any help on what im meant to do for all these pull requests is appreciated, bit new here but basic understanding of things

**Comment by patrickread** [comment]
@mikehow1984 I was actually developing pretty much the same thing concurrently! 😄 
Here's mine: https://github.com/patrickread/PokemonGo-Map/tree/alerts

I borrowed some SMTP code from yours, since I hadn't done that part yet. It uses a separate thread for the email alerting.

File: pogom/utils.py (hunk 1 above)
Comment (mikehow1984) [inline pogom/utils.py]: ah, the "example" in the name should've tipped me off

**Comment by MonkeyLeeT** [comment]
@patrickread @mikehow1984 Closing this one and expecting a new one having both of your effort.

**Comment by Sejadis** [comment]
time calculation for the mail doesnt work properly
im at utc +2 
expected output : i have till 4.13 AM
observerd output: i have till 7.13 AM

clicking the map link on my android phone opens the maps app but doenst show the location because "unsupported link"

**Comment by patrickread** [comment]
As @MonkeyLeeT requested, new PR available here: https://github.com/AHAAAAAAA/PokemonGo-Map/pull/1106

@Sejadis Interesting, link worked for me on iOS, consumed by Google Maps app. I think the link I have on my PR is the same as @mikehow1984's. Feel free to check the results from that PR on Android, don't have a device handy.

**Comment by xxzone** [comment]
This is awesome, thank you for this. The only issue I've had is it not keeping track of which ones it's already sent, so you get an email every loop through the steps. 

I'm glad I have it at all though, thanks again.

**Pull request closed without merging**